### PR TITLE
new jjob for ocean analysis bmat

### PR DESCRIPTION
**Pull request by AndrewEichmann-NOAA**
**Description**

Adds jjob for ocean analysis bmat. The intent is to have separate tasks for the ocean analysis and generation of the b matrix.

Addresses https://github.com/NOAA-EMC/GDASApp/issues/263

Once this is merged, additional modifications will be made to GDASApp to allow it to be run with the other global-workflow components of ocean/ice DA

**Type of change**

- [ ] New feature (non-breaking change which adds functionality)

**How Has This Been Tested?**

Tested on Hera with GDASApp branch feature/add-soca-bmat-jjob, produces output similar to stand-alone GDASApp ocean bmat script.

  
**Checklist**

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing tests pass with my changes

### CODE REVIEW

**Comment by AndrewEichmann-NOAA** [comment]
This PR includes only a single shell scripts, so not sure where the python test failure is coming from

**Comment by AndrewEichmann-NOAA** [comment]
@guillaumevernieres Who usually reviews this sort of thing?

**Review by guillaumevernieres — APPROVED** [review]
Looks good. @aerorahul and @WalterKolczynski-NOAA : This is PR 1 of 2 (or 3), that commit is none breaking and the added j-job is not used yet.

File: jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_BMAT [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    [[ ${status} -ne 0 ]] && exit "${status}"
+done
+
+
+##########################################
+# Source machine runtime environment
+##########################################
+. "${HOMEgfs}/env/${machine}.env" ocnanalrun
+status=$?
+[[ ${status} -ne 0 ]] && exit "${status}"
```
Comment (aerorahul) [inline jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_BMAT]:
```suggestion
export DATA="${DATAROOT}/${RUN}ocnanal_${cyc}"
source "${HOMEgfs}/ush/jjob_header.sh" -e "ocnanalrun" -c "base ocnanal ocnanalrun"
```

**Review by aerorahul — CHANGES_REQUESTED** [review]
This looks good.
However, we recently introduced a `jjob_header.sh`.

I have incorporated that as a suggested commit here.

File: jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_BMAT (hunk 1 above)
Comment (AndrewEichmann-NOAA) [reply jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_BMAT]: @aerorahul I'll take your word for it

**Review by aerorahul — APPROVED** [review]
looks good.

File: jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_BMAT [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+export DATA="${DATAROOT}/${RUN}ocnanal_${cyc}"
+source "${HOMEgfs}/ush/jjob_header.sh" -e "ocnanalrun" -c "base ocnanal ocnanalrun"
+
+
+##############################################
+# Set variables used in the script
+##############################################
+
+export CDUMP=${CDUMP:-${RUN:-"gfs"}}
+export COMPONENT="ocean"
```
Comment (aerorahul) [inline jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_BMAT]: Is it necessary to define and use this variable.
It is clear from the name of this script, this is an `OCEAN` component.

File: jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_BMAT [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+##############################################
+# Begin JOB SPECIFIC work
+##############################################
+
+export COMOUT=${COMOUT:-${ROTDIR}/${CDUMP}.${PDY}/${cyc}/${COMPONENT}}
```
Comment (aerorahul) [inline jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_BMAT]:
```suggestion
export COMOUT=${COMOUT:-${ROTDIR}/${CDUMP}.${PDY}/${cyc}/ocean}
```

File: jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_BMAT (hunk 2 above)
Comment (aerorahul) [inline jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_BMAT]:
```suggestion
```

File: jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_BMAT [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+export DATA="${DATAROOT}/${RUN}ocnanal_${cyc}"
+source "${HOMEgfs}/ush/jjob_header.sh" -e "ocnanalrun" -c "base ocnanal ocnanalrun"
+
+
+##############################################
+# Set variables used in the script
+##############################################
+
+export CDUMP=${CDUMP:-${RUN:-"gfs"}}
```
Comment (aerorahul) [inline jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_BMAT]:
```suggestion
```
This cannot be `gfs`, as the script is explicitly a `JGDAS` script.
Besides, both, `RUN` and `CDUMP` are defined upstream (has to), so this line accomplishes nothing.

**Review by aerorahul — COMMENTED** [review]
a couple of post-review comments to save time and effort later.

File: jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_BMAT (hunk 2 above)
Comment (AndrewEichmann-NOAA) [reply jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_BMAT]: revision committed

**Comment by AndrewEichmann-NOAA** [comment]
Thanks for the suggestions, @aerorahul

**Review by aerorahul — APPROVED** [review]
looks good.
thanks for the updates.

**Review by WalterKolczynski-NOAA — APPROVED** [review]

**Pull request merged**